website_sale_product_minimal_price/views/templates.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Sergio Teruel
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <template id="variants" inherit_id="website_sale.variants">
+        <!-- Sort product attribute values by minimal price extra -->
+        <xpath expr="//t[@t-if=&quot;variant_id.attribute_id.type == &apos;select&apos;&quot;]//t[@t-foreach='variant_id.value_ids']" position="attributes">
+            <attribute name="t-foreach">variant_id.value_ids.sorted(key=lambda x: x.price_extra)</attribute>
+        </xpath>
+        <xpath expr="//t[@t-if=&quot;variant_id.attribute_id.type == &apos;radio&apos;&quot;]//t[@t-foreach='variant_id.value_ids']" position="attributes">
+            <attribute name="t-foreach">variant_id.value_ids.sorted(key=lambda x: x.price_extra)</attribute>
+        </xpath>
+        <xpath expr="//li[@t-foreach='variant_id.value_ids']" position="attributes">
+            <attribute name="t-foreach">variant_id.value_ids.sorted(key=lambda x: x.price_extra)</attribute>
+        </xpath>
+    </template>
+
+    <template id="products_item" inherit_id="website_sale.products_item">
+        <!-- Display "From" if variants have distinct extra prices -->
+        <xpath expr="//span[@t-esc='product.website_price']" position="before">
+            <t t-if="product.product_variant_count &gt; 1 and len(set(product.product_variant_ids.mapped('price_extra'))) != 1">
```

Comment:
For achieving this, maybe you need an extra computed stored field at template level for indicating if there are different prices.